### PR TITLE
Bus str onboarding

### PR DESCRIPTION
**Pull request by FarazA22**
### Ticket

[#184859520](https://www.pivotaltracker.com/story/show/184859520)

### Steps to Test

- Log in as a new user, on the dashboard the progress card is updated to count the progress for the plan section only. A prompt in the start section navigates you to the business structure task.
- The profile page has an update to the business structure section. It's similar to the NAICS code functionality. The add button routes you to the business structure task.
- Navigate to the business structure task, select and save the business structure. You have the option to edit and remove the saved business structure. The roadmap is displaying a prompt until you saved a selection. If you are on another task, there is a link that navigates you to the business structure task, the link does not exist when you are on the task.
- The profile page displays an edit link when the business structure is saved. 
- The progress card is updated to count the progress for both sections when the business structure is saved. 
- In formation, the edit link takes you to the business structure task.
- When the business is formed, the business structure is locked in the profile and in the business structure task.

## Code author checklist

- [x] I have performed a self-review of my code
- [x] I have created and/or updated relevant documentation, if necessary
- [x] I have not used any relative imports
- [x] I have checked for and removed instances of unused config from CMS
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config

### CODE REVIEW

File: shared/src/test/factories.ts [hunk 1]
```diff
@@ -221,7 +221,7 @@ export const generateProfileData = (
     responsibleOwnerName: `some-responsible-owner-name-${randomInt()}`,
     tradeName: `some-trade-name-${randomInt()}`,
     industryId: industry.id,
-    legalStructureId: randomLegalStructure().id,
+    legalStructureId: undefined,
```
Comment (aloverso) [inline shared/src/test/factories.ts:224]: Why this change? A generator isn't necessarily representing "initial" state, it's representing a random state, so changing the onboarding doesn't seem like this should change?

File: web/cypress/e2e/business-formation.spec.ts [hunk 2]
```diff
@@ -59,14 +62,15 @@ describe("Business Formation [feature] [all] [group2]", () => {
 
     completeNewBusinessOnboarding({
       industry,
-      legalStructureId,
     });
+    completeBusinessStructureTask({ legalStructureId });
+
+    updateNewBusinessProfilePage({ townDisplayName });
```
Comment (aloverso) [inline web/cypress/e2e/business-formation.spec.ts:68]: Why this change? It doesn't seem related to the business structure?

Comment (aloverso) [reply web/cypress/e2e/business-formation.spec.ts:68]: Same question with the address removals below

File: web/cypress/e2e/check-license-status.spec.ts [hunk 3]
```diff
@@ -10,15 +10,16 @@ describe("check license status [feature] [all] [group1]", () => {
...
     const industry = LookupIndustryById("home-contractor");
-    const legalStructureId = "limited-liability-company";
+    const legalStructureId = "general-partnership";
+    const townDisplayName = "Barnegat";
+    const businessName = "Aculyst";
 
     completeNewBusinessOnboarding({
       industry,
-      legalStructureId,
     });
```
Comment (aloverso) [inline web/cypress/e2e/check-license-status.spec.ts:18]: As a style not, now that this call is so short I'd probably one-line it instead of three lines

File: web/cypress/e2e/dashboard.spec.ts [hunk 4]
```diff
@@ -103,13 +119,18 @@ describe("Dashboard [feature] [all] [group2]", () => {
 
       it("update the industry and verifies the dashboard tasks are updated", () => {
         const industry = LookupIndustryById("e-commerce");
-        const legalStructureId = "limited-liability-company";
+        const legalStructureId = "general-partnership";
+        const townDisplayName = "Barnegat";
 
         completeNewBusinessOnboarding({
           industry,
           legalStructureId,
```
Comment (aloverso) [inline web/cypress/e2e/dashboard.spec.ts:131]: Is this an oversight? Should legal structure be removed

File: web/cypress/e2e/dashboard.spec.ts [hunk 5]
```diff
@@ -135,12 +156,18 @@ describe("Dashboard [feature] [all] [group2]", () => {
       it("open and closes contextual info panel on get EIN from the IRS Task screen", () => {
         const industry = LookupIndustryById("e-commerce");
         const legalStructureId = "general-partnership";
+        const townDisplayName = "Barnegat";
 
         completeNewBusinessOnboarding({
           industry,
           legalStructureId,
```
Comment (aloverso) [inline web/cypress/e2e/dashboard.spec.ts:168]: Also here?

Comment (aloverso) [reply web/cypress/e2e/dashboard.spec.ts:168]: I wonder why typecheck didnt flag this

File: web/cypress/e2e/deferred-onboarding.spec.ts [hunk 6]
```diff
@@ -24,6 +26,7 @@ describe("Deferred Onboarding [feature] [all] [group5]", () => {
     describe("onboarded as STARTING - PublicFiling", () => {
       const testLocationInThreePlaces = (): void => {
         it("can provide location in Location-Dependent task", () => {
+          completeBusinessStructureTask({ legalStructureId: randomPublicFilingLegalStructure() });
```
Comment (aloverso) [inline web/cypress/e2e/deferred-onboarding.spec.ts]: Why do these (here and in the tests below) instead of in the before each?

File: web/cypress/support/helpers.ts [hunk 7]
```diff
@@ -590,12 +487,24 @@ export const checkNewBusinessProfilePage = ({
     onProfilePage.getHomeBased(!homeBasedQuestion).should("not.be.checked");
   }
 
-  onProfilePage
-    .getLegalStructure()
-    .parent()
-    .find("input")
-    .invoke("prop", "value")
-    .should("contain", companyType);
+  if (companyType !== undefined) {
```
Comment (aloverso) [inline web/cypress/support/helpers.ts]: I see we were using `companyType` instead of `legalStructure` - it seems preferable to use our domain language and not something new. We could refactor this

File: web/cypress/support/helpers.ts [hunk 8]
```diff
@@ -861,6 +770,18 @@ export const updateExistingBusinessProfilePage = ({
   cy.url().should("contain", "/dashboard");
 };
 
+export const completeBusinessStructureTask = ({ legalStructureId }: { legalStructureId: string }): void => {
+  cy.get('[data-task="business-structure"]').click();
+  cy.url().should("contain", "/business-structure");
+  cy.log(legalStructureId as string);
```
Comment (aloverso) [inline web/cypress/support/helpers.ts:752]: delete

File: web/cypress/support/helpers.ts [hunk 9]
```diff
@@ -861,6 +770,18 @@ export const updateExistingBusinessProfilePage = ({
   cy.url().should("contain", "/dashboard");
 };
 
+export const completeBusinessStructureTask = ({ legalStructureId }: { legalStructureId: string }): void => {
+  cy.get('[data-task="business-structure"]').click();
+  cy.url().should("contain", "/business-structure");
```
[the remark this answers is not in the document]
Comment (aloverso) [reply web/cypress/support/helpers.ts:751]: to wait for the page to load, look for something else instead (test id on the page?)

File: web/cypress/support/page_objects/onboardingPage.ts [hunk 10]
```diff
@@ -83,7 +83,7 @@ export class OnboardingPage {
   }
 
   getLegalStructure(companyType: string) {
-    return cy.get(`[data-value="${companyType}"]`);
+    return cy.get(`input[name="legal-structure"][value="${companyType}"]`);
```
Comment (aloverso) [inline web/cypress/support/page_objects/onboardingPage.ts:86]: do we need this function, if we never interact with legal structure on the onboarding page anymore?

File: web/cypress/support/page_objects/onboardingPage.ts [hunk 11]
```diff
@@ -179,7 +179,9 @@ export class OnboardingPage {
   }
 
   selectLegalStructure(companyType: string) {
-    this.getLegalStructure(companyType).click({ force: true });
+    this.getLegalStructure(companyType).scrollIntoView();
+    this.getLegalStructure(companyType).check();
+    this.getLegalStructure(companyType).check();
```
Comment (aloverso) [inline web/cypress/support/page_objects/onboardingPage.ts:184]: do we need this function, if we never interact with legal structure on the onboarding page anymore?

File: web/src/components/filings-calendar/tax-access-modal/TaxAccessModal.test.tsx [hunk 12]
```diff
@@ -36,6 +40,7 @@ describe("<TaxAccessModal />", () => {
         generateUserData({
           profileData: generateProfileData({
             businessPersona: persona as BusinessPersona,
+            legalStructureId: randomLegalStructure().id,
```
Comment (aloverso) [inline web/src/components/filings-calendar/tax-access-modal/TaxAccessModal.test.tsx:43]: This wouldn't be necessary if we hadnt set it to undefined in the generators

I think it's the reasonable assumption that the generators set a value in pretty much every field

File: web/test/pages/onboarding/onboarding-owning.test.tsx [hunk 13]
```diff
@@ -216,6 +217,7 @@ describe("onboarding - owning a business", () => {
       profileData: generateProfileData({
         businessPersona: "OWNING",
         operatingPhase: "GUEST_MODE_OWNING",
+        legalStructureId: randomLegalStructure().id,
```
Comment (aloverso) [inline web/test/pages/onboarding/onboarding-owning.test.tsx:220]: Regardless of the change to factories, this seems unneeded because oscar also does not select business structure in onboarding anyways

File: web/test/pages/onboarding/onboarding-starting.test.tsx [hunk 14]
```diff
@@ -318,22 +276,7 @@ describe("onboarding - starting a business", () => {
     });
   });
 
-  it("removes required fields error when user goes back", async () => {
```
Comment (aloverso) [inline web/test/pages/onboarding/onboarding-starting.test.tsx:321]: This test isn't necessarily testing legal structure, it's testing banner behavior and using legal structure as an example. 

I think we need to keep this test but just use a different banner error (if there is one?)

File: web/test/pages/profile.test.tsx [hunk 15]
```diff
@@ -250,7 +253,10 @@ describe("profile", () => {
 
     beforeEach(() => {
       userData = generateUserData({
-        profileData: generateProfileData({ businessPersona: "STARTING" }),
+        profileData: generateProfileData({
+          businessPersona: "STARTING",
+          legalStructureId: randomLegalStructure().id,
```
Comment (aloverso) [inline web/test/pages/profile.test.tsx:258]: same comment - not needed if we dont change the factory

File: web/test/pages/profile.test.tsx [hunk 16]
```diff
@@ -305,7 +311,11 @@ describe("profile", () => {
     describe("formation date", () => {
       it("does not display when empty", () => {
         const initialUserData = generateUserData({
-          profileData: generateProfileData({ businessPersona: "STARTING", dateOfFormation: "" }),
+          profileData: generateProfileData({
+            businessPersona: "STARTING",
+            dateOfFormation: "",
+            legalStructureId: randomLegalStructure().id,
```
Comment (aloverso) [inline web/test/pages/profile.test.tsx:356]: same comment - not needed if we dont change the factory

Comment (aloverso) [reply web/test/pages/profile.test.tsx:356]: (in this whole file)

File: api/src/domain/user/updateOperatingPhase.test.ts [hunk 17]
```diff
@@ -29,50 +29,84 @@ describe("updateOperatingPhase", () => {
...
-          operatingPhase: "GUEST_MODE",
+          operatingPhase: "NEEDS_BUSINESS_STRUCTURE",
           legalStructureId: "general-partnership",
         }),
+        taskProgress: { [businessStructureTaskId]: "COMPLETED" },
       });
       expect(updateOperatingPhase(userData).profileData.operatingPhase).toBe("NEEDS_TO_REGISTER_FOR_TAXES");
     });
+
+    it("updates status to FORMED_AND_REGISTERED when formation and business structure tasks are completed and PublicFiling legal structure and tax task is complete", () => {
```
Comment (aloverso) [inline api/src/domain/user/updateOperatingPhase.test.ts:68]: Is there a missing corresponding one here for trade name businesses?

File: api/src/domain/user/updateOperatingPhase.test.ts [hunk 18]
```diff
@@ -84,20 +118,40 @@ describe("updateOperatingPhase", () => {
           operatingPhase: "NEEDS_TO_FORM",
           legalStructureId: "general-partnership",
         }),
-        taskProgress: { [formationTaskId]: "IN_PROGRESS" },
+        taskProgress: { [formationTaskId]: "IN_PROGRESS", [businessStructureTaskId]: "COMPLETED" },
```
Comment (aloverso) [inline api/src/domain/user/updateOperatingPhase.test.ts:121]: I think this outcome is depenedent on the Tax Task, not the formation task (since this is for trade name)

even if the formation task was complete here, the outcome would be the same. 
I think the test should make that explicit in the test name (that it's the tax task being incomplete that makes it needs-to-register)

File: api/src/domain/user/updateOperatingPhase.test.ts [hunk 19]
```diff
@@ -84,20 +118,40 @@ describe("updateOperatingPhase", () => {
...
       });
       expect(updateOperatingPhase(userData).profileData.operatingPhase).toBe("NEEDS_TO_REGISTER_FOR_TAXES");
     });
+
+    it("updates status back to NEEDS_BUSINESS_STRUCTURE if business structure task is not completed", () => {
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          businessPersona: "STARTING",
+          operatingPhase: "NEEDS_TO_FORM",
+          legalStructureId: "limited-liability-company",
```
Comment (aloverso) [inline api/src/domain/user/updateOperatingPhase.test.ts:131]: should we either have another test for trade name, or do a it.each or a randomizer here?

File: api/src/domain/user/updateOperatingPhase.test.ts [hunk 20]
```diff
@@ -84,20 +118,40 @@ describe("updateOperatingPhase", () => {
...
+          [taxTaskId]: "COMPLETED",
+        },
+      });
+      expect(updateOperatingPhase(userData).profileData.operatingPhase).toBe("NEEDS_BUSINESS_STRUCTURE");
+    });
   });
 
   describe("NEEDS_TO_REGISTER_FOR_TAXES", () => {
-    it("updates status to FORMED_AND_REGISTERED when tax task and formation task is completed", () => {
+    it("updates status to FORMED_AND_REGISTERED when tax, formation, and business structure task are completed", () => {
```
Comment (aloverso) [inline api/src/domain/user/updateOperatingPhase.test.ts:144]: i wonder if we could simplify these tests to be more declarative now that we did that amazing refactor to simplify the code to be more declarative

Comment (aloverso) [reply api/src/domain/user/updateOperatingPhase.test.ts:144]: as a follow up refactor, doesn't need to happen before merging code

File: web/cypress/e2e/guest.spec.ts [hunk 21]
```diff
@@ -9,7 +9,8 @@ import {
   updateNewBusinessProfilePage,
 } from "../support/helpers";
 
-describe("Guest Dashboard [feature] [all] [group2]", () => {
+// skipping until there is clarity on how save business structure button should behave for guest mode
```
Comment (aloverso) [inline web/cypress/e2e/guest.spec.ts]: did we get this clarity?

File: web/cypress/support/page_objects/dashboardPage.ts [hunk 22]
```diff
@@ -30,6 +30,7 @@ export class DashboardPage {
 
   selectHomeBased(radio: boolean) {
     this.getHomeBased(radio).check();
+    this.getHomeBased(radio).check();
```
Comment (aloverso) [inline web/cypress/support/page_objects/dashboardPage.ts:33]: Why double clicking the same button?

Comment (aloverso) [reply web/cypress/support/page_objects/dashboardPage.ts:33]: could leave a comment (!) here to explain

File: web/src/pages/profile.tsx [hunk 23]
```diff
@@ -113,11 +110,6 @@ const ProfilePage = (props: Props): ReactElement => {
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const redirect = (params?: { [key: string]: any }, routerType = router.push): Promise<boolean> => {
-    if (checkQueryValue(router, QUERIES.path, "businessFormation")) {
-      const formationUrlSlug = getTaskFromRoadmap(roadmap, formationTaskId)?.urlSlug ?? "";
```
Comment (aloverso) [inline web/src/pages/profile.tsx:117]: I think we should be using this getTaskFromRoadmap function in the code above that changed in BusinessStep and such. That way, the URL can get changed by the content team, and the code will still work because it's dynamically looking up the url from the roadmap

File: web/src/components/tasks/BusinessStructureTask.test.tsx [hunk 24]
```diff
@@ -192,4 +192,16 @@ describe("<BusinessStructureTask />", () => {
       expect(currentUserData().taskProgress[taskId]).toEqual("COMPLETED");
     });
   });
+
+  it("updates operating phase from GUEST_MODE to GUEST_MODE_WITH_BUSINESS_STRUCTURE when business structure is saved", async () => {
```
Comment (aloverso) [inline web/src/components/tasks/BusinessStructureTask.test.tsx:196]: I'd love a corresponding test here that it does NOT update to GUEST_MODE_WITH_BUSINESS_STRUCTURE when the user is not in GUEST_MODE

Right now, if I deleted the ternary in the OnboardingLegalStructure and just had it update to GUEST_MODE_WITH_BUSINESS_STRUCTURE always, then this test would still pass and nothing would fail

File: web/src/components/dashboard/BusinessStructurePrompt.test.tsx [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+  it("renders notCompletedTaskPromptAnyTask from config", () => {
+    useMockRouter({
+      asPath: "any",
+    });
+    render(<BusinessStructurePrompt />);
+    expect(screen.getByTestId("any-task-content")).toBeInTheDocument();
+    expect(screen.queryByTestId("business-structure-task-content")).not.toBeInTheDocument();
+  });
+
+  it("renders notCompletedTaskPromptBusinessStructureTask from config", () => {
```
Comment (aloverso) [inline web/src/components/dashboard/BusinessStructurePrompt.test.tsx:38]: it would be helpful for these test names to include the "when" because otherwise they read as conflicting to the previous statement

"hides business structure prompt button when isButtonHidden is true"
"hides business structure prompt button when isButtonHidden is false or not included"
"renders generic config when on any task"
"renders specialized config when on business structure task"

File: web/src/components/dashboard/BusinessStructurePrompt.tsx [hunk 26]
```diff
@@ -19,20 +22,31 @@ export const BusinessStructurePrompt = (): ReactElement => {
...
-            isColor="primary"
-            isRightMarginRemoved
-            isFullWidthOnDesktop
-            onClick={(): void => {
-              router.push(ROUTES.businessStructureTask);
-            }}
-          >
-            {Config.businessStructurePrompt.buttonText}
-          </PrimaryButton>
+          {router.asPath === ROUTES.businessStructureTask ? (
```
Comment (aloverso) [inline web/src/components/dashboard/BusinessStructurePrompt.tsx:25]: Same comment as before - we should look up the url from the roadmap instead of hardcoding it.

That'll have to be a change to the test as well.

File: web/src/components/dashboard/BusinessStructurePrompt.test.tsx [hunk 27]
```diff
@@ -0,0 +1,52 @@
...
+    render(<BusinessStructurePrompt />);
+    expect(screen.getByTestId("business-structure-prompt-button")).toBeInTheDocument();
+  });
+
+  it("renders notCompletedTaskPromptAnyTask from config", () => {
+    useMockRouter({
+      asPath: "any",
+    });
+    render(<BusinessStructurePrompt />);
+    expect(screen.getByTestId("any-task-content")).toBeInTheDocument();
```
Comment (aloverso) [inline web/src/components/dashboard/BusinessStructurePrompt.test.tsx:34]: This test isn't actually testing the config, which makes the test name a bit misleading. I'm fine with using the testid to make it easier to test for the markdown string, but maybe just changing the test name to match.

File: web/src/components/roadmap/MiniRoadmap.tsx [hunk 28]
```diff
@@ -44,32 +50,65 @@ export const MiniRoadmap = (props: Props): ReactElement => {
...
+
+  const renderStartSection = (
+    <SectionAccordion
+      key={sectionNamesInRoadmap[1]}
+      sectionType={sectionNamesInRoadmap[1]}
+      mini={true}
+      isDividerDisabled={displayBusinessStructurePrompt}
+    >
+      {displayBusinessStructurePrompt ? (
+        <BusinessStructurePrompt isButtonHidden={true} />
```
Comment (aloverso) [inline web/src/components/roadmap/MiniRoadmap.tsx:84]: It seems like this is never being set with isButtonHidden = false?

Where is the logic that changes this prompt based on if we're on the task or not?

File: content/src/fieldConfig/profile.json [hunk 29]
```diff
@@ -238,6 +238,9 @@
           "headerContextualInfo": "business-structure-learn-more",
           "description": "",
           "errorTextRequired": "Select a business structure.",
+          "editText": "Edit",
+          "addText": "Add",
+          "notEnteredText": "*Not entered*",
```
Comment (aloverso) [inline content/src/fieldConfig/profile.json:243]: This should be handled in code with <em> instead of in content

File: web/cypress/e2e/profile.spec.ts [hunk 30]
```diff
@@ -28,10 +22,9 @@ describe("Profile [feature] [all] [group1]", () => {
     cy.loginByCognitoApi();
   });
 
-  describe("navigates to profile page and updates all fields", () => {
+  describe.only("navigates to profile page and updates all fields", () => {
```
Comment (aloverso) [inline web/cypress/e2e/profile.spec.ts:25]: debugger should be removed?

File: web/src/components/profile/ProfileBusinessStructure.test.tsx [hunk 31]
```diff
@@ -0,0 +1,71 @@
...
+jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
+const Config = getMergedConfig();
+
+const renderComponent = (profileData: Partial<ProfileData>): void => {
+  const data = generateProfileData({
+    businessPersona: "STARTING",
+    ...profileData,
+  });
+  render(
+    <ProfileDataContext.Provider
```
Comment (aloverso) [inline web/src/components/profile/ProfileBusinessStructure.test.tsx:19]: You should not need to do this yourself - this is what the `WithStatefulProfileData` helper does already

File: web/src/components/profile/ProfileBusinessStructure.test.tsx [hunk 32]
```diff
@@ -0,0 +1,71 @@
...
+    renderComponent({ legalStructureId: "corporation" });
+    expect(screen.getByText(configForField.editText)).toHaveAttribute(
+      "href",
+      "/tasks/some-business-structure-url"
+    );
+  });
+
+  it("displays Not Entered text when user has no NAICS code", () => {
+    renderComponent({ legalStructureId: undefined });
+    expect(screen.getByTestId("not-entered")).toBeInTheDocument();
```
Comment (aloverso) [inline web/src/components/profile/ProfileBusinessStructure.test.tsx:63]: This can be changed to a getByText when the config change I pointed out is made

File: web/src/components/profile/ProfileBusinessStructure.tsx [hunk 33]
```diff
@@ -0,0 +1,72 @@
...
+      <div className="flex flex-row">
+        <FieldLabelProfile fieldName="legalStructureId" />
+        {isFormed ? (
+          <div className="margin-left-2">
+            <ArrowTooltip
+              title={Config.profileDefaults.lockedFieldTooltipText}
+              data-testid="business-structure-tooltip"
+            >
+              <div className="fdr fac font-body-lg">
+                <Icon>help_outline</Icon>
```
Comment (aloverso) [inline web/src/components/profile/ProfileBusinessStructure.tsx:46]: This is copy-pasted from FieldLabelProfile - could we make use of the `locked` prop on that instead of rewriting it here?

ie would this work?

```
<FieldLabelProfile fieldName="legalStructureId" locked={isFormed}/>
{!isFormed && <a
            className="margin-left-2"
            href={businessStructureTaskUrl}
            data-testid={"business-structure-task-link"}
          >
            {state.profileData.legalStructureId ? contentFromConfig.editText : contentFromConfig.addText}
          </a>
}
```

File: web/src/lib/domain-logic/isGetFilingResponseSuccussful.ts [hunk 34]
```diff
@@ -0,0 +1,5 @@
+import { UserData } from "@businessnjgovnavigator/shared/";
+
+export const isGetFilingResponseSuccussful = (userData: UserData | undefined): boolean => {
```
Comment (aloverso) [inline web/src/lib/domain-logic/isGetFilingResponseSuccussful.ts:3]: This already exists in the form of `hasCompletedFormation` function

File: web/src/components/dashboard/SidebarCardTaskProgress.tsx [hunk 35]
```diff
@@ -33,9 +35,13 @@ export const SidebarCardTaskProgress = (props: Props): ReactElement => {
   const { Config } = useConfig();
   const { userData } = useUserData();
   const { roadmap } = useRoadmap();
+  const modifiedRoadmap = LookupOperatingPhaseById(userData?.profileData.operatingPhase)
+    .displayBusinessStructurePrompt
+    ? roadmapWithSectionSpecificTasks(roadmap as Roadmap, "PLAN")
```
Comment (aloverso) [inline web/src/components/dashboard/SidebarCardTaskProgress.tsx:40]: I would rather make this a function or memo and add guard clauses if it is having issues with unfeined. Or just have your function allow and handle undefined values. Using `as` basically wipes away *any* typechecking that typescript would give us, and can lead to completely hiding away errors that SHOULD be occurring if the types change.

File: web/src/lib/domain-logic/roadmapWithSectionSpecificTasks.ts [hunk 36]
```diff
@@ -0,0 +1,22 @@
+import { Roadmap, Step, Task } from "@/lib/types/types";
+import { SectionType } from "@businessnjgovnavigator/shared/userData";
+import { cloneDeep } from "lodash";
+
+export const roadmapWithSectionSpecificTasks = (roadmap: Roadmap, section: SectionType): Roadmap => {
+  const result = { steps: [], tasks: [] } as { steps: Step[]; tasks: Task[] };
```
Comment (aloverso) [inline web/src/lib/domain-logic/roadmapWithSectionSpecificTasks.ts:6]: Again, we should avoid the `as`.

Can we do `const result: Roadmap = { steps: [], tasks: [] }` ?

File: shared/src/test/factories.ts (hunk 1 above)
Comment (FarazA22) [reply shared/src/test/factories.ts:224]: agree, will revert

File: web/src/components/filings-calendar/tax-access-modal/TaxAccessModal.test.tsx (hunk 12 above)
Comment (FarazA22) [reply web/src/components/filings-calendar/tax-access-modal/TaxAccessModal.test.tsx:43]: agree will remove

File: web/src/components/onboarding/OnboardingFlows.tsx [hunk 37]
```diff
@@ -57,14 +56,6 @@ export const onboardingFlows: Record<FlowType, OnboardingFlow> = {
           </>
         ),
       },
-      {
-        component: (
-          <>
-            <FieldLabelOnboarding fieldName="legalStructureId" />
-            <OnboardingLegalStructure<OnboardingErrors> errorTypes={["REQUIRED_LEGAL"]} />
```
[the remark this answers is not in the document]
Comment (FarazA22) [reply web/src/components/onboarding/OnboardingFlows.tsx:64]: agree

File: web/cypress/e2e/guest.spec.ts (hunk 21 above)
Comment (FarazA22) [reply web/cypress/e2e/guest.spec.ts]: Yes
